src/Components/SoundWave/index.js

```diff
@@ -0,0 +1,38 @@
+import React, { useRef, useEffect, useState } from "react";
+import PropTypes from "prop-types";
+
+function SoundWave({ src, waveColor, progressColor }) {
+  const waveFormRef = useRef(null);
+  const waveSurferRef = useRef(null);
+  const [isPlay, setIsPlay] = useState(false);
+
+  useEffect(() => {
+    waveSurferRef.current = window.WaveSurfer.create({
+      container: waveFormRef.current,
+      waveColor,
+      progressColor
+    });
+    waveSurferRef.current.load(src);
+    return () => waveSurferRef.current.destroy();
+  }, []);
+
+  const handlePlayAndPause = () => {
+    waveSurferRef.current.playPause();
+    setIsPlay(isPlay === true ? false : true);
```

Comment:
We can change this line to `setIsPlay(!isPlay)` to clean things up a bit.

- [ ] simplify line